pydicom/valuerep.py

```diff
@@ -43,6 +43,312 @@
 # (the component separator '=' is handled separately)
 PN_DELIMS = {0x5e}
 
+# maximum allowed value length for string VRs
+# VRs with a maximum length of 2^32 (UR and UT) are not checked
+MAX_VALUE_LEN = {
+    "AE": 16,
+    "CS": 16,
+    "DS": 16,
+    "IS": 12,
+    "LO": 64,
+    "LT": 10240,
+    "SH": 16,
+    "ST": 1024,
+    "UI": 64
+}
+
+
+def _range_regex(regex: str) -> str:
+    """Compose a regex that allows ranges of the given regex,
+    as defined for VRs DA, DT and TM in PS 3.4, C.2.2.2.5.
+    """
+    return fr"^{regex}$|^\-{regex} ?$|^{regex}\- ?$|^{regex}\-{regex} ?$"
+
+
+# regular expressions to match valid values for some VRs
+AE_REGEX = re.compile(r"^[\x20-\x7e]*$")
+AS_REGEX = re.compile(r"^\d\d\d[DWMY]$")
+CS_REGEX = re.compile(r"^[A-Z0-9 _]*$")
+DS_REGEX = re.compile(
+    r"^ *[+\-]?(\d+|\d+\.\d*|\.\d+)([eE][+\-]?\d+)? *$"
+)
+IS_REGEX = re.compile(r"^ *[+\-]?\d+ *$")
+DA_REGEX = re.compile(_range_regex(r"\d{4}(0[1-9]|1[0-2])([0-2]\d|3[01])"))
+DT_REGEX = re.compile(_range_regex(
+    r"\d{4}((0[1-9]|1[0-2])(([0-2]\d|3[01])(([01]\d|2[0-3])"
+    r"([0-6]\d([0-5]\d(\.\d{1,6} ?)?)?)?)?)?)?([+-][01]\d\d\d)?")
+)
+TM_REGEX = re.compile(_range_regex(
+    r"([01]\d|2[0-3])([0-6]\d([0-5]\d(\.\d{1,6} ?)?)?)?"))
```

Comment:
Matches `006100`, maybe `([01]\d|2[0-3])(6[0]|[0-5]\d([0-5]\d(\.\d{1,6} ?)?)?)?`

Reply:
`DT_REGEX` will also match an minute in 00-69, and `0000aa` will pass the `TM_REGEX` check, while `000060` fails.

Also, should it be the seconds in the range 0-60, rather than the minute?